File: public/examples/integrations/mcp-servers/box_api/download_file_content_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "BoxApi.DownloadFileContent"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'file_identifier': 'f_9a8b7c6d',
    'file_version_to_download': 'v2',
    'optional_access_token': 'rk-read-12345',
    'download_byte_range': 'bytes=0-1048575',
    'shared_link_with_optional_password': 'shared_link=https://files.example.com/s/abc123&shared_link_password=secret'
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/box_api/download_file_content_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "BoxApi.DownloadFileContent";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "file_identifier": "f_9a8b7c6d",
  "file_version_to_download": "v2",
  "optional_access_token": "rk-read-12345",
  "download_byte_range": "bytes=0-1048575",
  "shared_link_with_optional_password": "shared_link=https://files.example.com/s/abc123&shared_link_password=secret"
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
